File: server/datasets/python_source-code/Multiple-G1/renamed_48_bst.py
class Node_12:
    def __init__(self, val_12):
        self.val_12 = val_12
        self.lhs_12 = None
        self.rhs_12 = None

class Tree_12:
    def __init__(self):
        self.rt_12 = None

    def ins_12(self, val_12):
        if self.rt_12 is None:
            self.rt_12 = Node_12(val_12)
        else:
            self._ins_12_rec(self.rt_12, val_12)

    def _ins_12_rec(self, nd_12, val_12):
        if val_12 < nd_12.val_12:
            if nd_12.lhs_12 is None:
                nd_12.lhs_12 = Node_12(val_12)
            else:
                self._ins_12_rec(nd_12.lhs_12, val_12)
        elif val_12 > nd_12.val_12:
            if nd_12.rhs_12 is None:
                nd_12.rhs_12 = Node_12(val_12)
            else:
                self._ins_12_rec(nd_12.rhs_12, val_12)

    def srch_12(self, val_12):
        return self._srch_12_rec(self.rt_12, val_12)

    def _srch_12_rec(self, nd_12, val_12):
        if nd_12 is None or nd_12.val_12 == val_12:
            return nd_12
        if val_12 < nd_12.val_12:
            return self._srch_12_rec(nd_12.lhs_12, val_12)
        return self._srch_12_rec(nd_12.rhs_12, val_12)

File: server/datasets/java_source-code/Multiple-G1/renamed_48_bst.java
class Node_12 {
    int val_12;
    Node_12 lhs_12, rhs_12;
    public Node_12(int item) {
        val_12 = item;
        lhs_12 = rhs_12 = null;
    }
}
public class Tree_12 {
    public static void main(String[] args) {
        Tree_12 bst = new Tree_12();
        bst.ins_12(50);
        bst.srch_12(50);
    }
    Node_12 rt_12;
    public void ins_12(int val_12) {
        rt_12 = in_rec_12(rt_12, val_12);
    }
    Node_12 in_rec_12(Node_12 rt_12, int val_12) {
        if (rt_12 == null) return new Node_12(val_12);
        if (val_12 < rt_12.val_12) rt_12.lhs_12 = in_rec_12(rt_12.lhs_12, val_12);
        else if (val_12 > rt_12.val_12) rt_12.rhs_12 = in_rec_12(rt_12.rhs_12, val_12);
        return rt_12;
    }
    Node_12 srch_12(int val_12) {
        return sr_rec_12(rt_12, val_12);
    }
    Node_12 sr_rec_12(Node_12 rt_12, int val_12) {
        if (rt_12 == null || rt_12.val_12 == val_12) return rt_12;
        if (rt_12.val_12 < val_12) return sr_rec_12(rt_12.rhs_12, val_12);
        return sr_rec_12(rt_12.lhs_12, val_12);
    }
}